### ISSUE TITLE
addr2line '': No such file

### ISSUE THREAD

**Issue by Rahmerh** [wee-slack/wee-slack#868]
I recently starting getting the following errors when trying to start wee-slack:

```
======= WeeChat backtrace ======= 
         │13:51:25     | (written by WeeChat 3.6, compiled on Jul 10 2022 14:32:15)
         │13:51:25     | 001  ??:0 [function ??]
         │13:51:25     | 002  ??:0 [function ??]
         │13:51:25     | addr2line: '0x660c4': No such file
         │13:51:25     | addr2line: '0x14ea14': No such file
         │13:51:25     | addr2line: /usr/lib/system/libsystem_platform.dylib: cannot get addresses from archive
         │13:51:25     | addr2line: '0x18234': No such file
         │13:51:25     | addr2line: '0x1847c': No such file
         │13:51:25     | addr2line: '0x1a2ad0': No such file
         │13:51:25     | addr2line: '0x3800c': No such file
         │13:51:25     | addr2line: '0x37e4c': No such file
         │13:51:25     | addr2line: '0x3661c': No such file
         │13:51:25     | addr2line: '0x362f8': No such file
         │13:51:25     | addr2line: '0x35e6c': No such file
         │13:51:25     | addr2line: '0x35da0': No such file
         │13:51:25     | addr2line: '0x35d4c': No such file
         │13:51:25     | addr2line: '0x2eb60': No such file
         │13:51:25     | addr2line: '0x1a33a0': No such file
         │13:51:25     | addr2line: '0x2e514': No such file
         │13:51:25     | addr2line: '0x2e430': No such file
         │13:51:25     | addr2line: '0x6b6c': No such file
         │13:51:25     | addr2line: '0x23b8': No such file
         │13:51:25     | addr2line: '0x15354': No such file
         │13:51:25     | addr2line: '0x1254c': No such file
         │13:51:25     | addr2line: '0x29a1c': No such file
         │13:51:25     | addr2line: '0x2975c': No such file
         │13:51:25     | addr2line: '0x206b0': No such file
```

I don't think I have changed anything on my side, I've installed the latest of binutils. I also removed my `~/.weechat` folder and reinstalled everything. Still happens on a fresh install.

I have weechat 3.6 installed with wee-slack latest

**Comment by trygveaa** [comment]
Hm, so the error about addr2line is an error while trying to display info about the original error. From the output here, it's impossible to know what the original error was. Could you try installing addr2line so we get more info about the original error?

**Comment by trygveaa** [comment]
Or maybe it's not that addr2line is missing, but that it fails to look up info about the error. If so, could you try following https://weechat.org/files/doc/stable/weechat_user.en.html#report_crashes

**Comment by Rahmerh** [comment]
It was missing before, but after re-install of binutils it found it, but then started throwing those errors.

I'll follow that guide and let you know

**Comment by Rahmerh** [comment]
When I try to build using cmake I get 'Php not found'. Any ideas?

**Comment by trygveaa** [comment]
You can drop building the PHP plugin with the `-DENABLE_PHP=OFF` option to cmake.

You can also try logging all the calls wee-slack makes to weechat to see what the last calls it makes before crashing are, which might give some clue. I've made a branch which logs the calls to `/tmp/weeslack-debug/weechat-calls.log` here: https://github.com/wee-slack/wee-slack/blob/log-weechat-calls/wee_slack.py

Note that it will log the Slack token in some of the lines, so don't post the whole log file uncensored here, but post the last lines and check that they don't include any secrets.

**Comment by Rahmerh** [comment]
So I think there are 2 interesting blocks, before and after the giant stacktrace about addr2line not finding a file.

Before the giant block it logs this:

```
Calling weechat function hook_timer with args (10, 0, 0, 'handle_next', ''), {}
 250   │ Calling weechat function bar_item_update with args ('slack_typing_notice',), {}
 251   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: from slow queue'), {}
 252   │ Calling weechat function bar_item_update with args ('slack_typing_notice',), {}
 253   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: from slow queue'), {}
 254   │ Calling weechat function bar_item_update with args ('slack_typing_notice',), {}
 255   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: from slow queue'), {}
 256   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: stored context VIG29OKHRTIMOPA33SZ57X2ZY6SYU5H9F4RE4U4S https://api.slack.com/api/rtm.start?batch_presence_aware=1&mpim_aware=true&token=xoxp-95559983717-38270282668
       │ 85-3976626606199-2d5f745149847c7ea8ffe925290796be '), {}
 257   │ Calling weechat function hook_process_hashtable with args ('url:', {'useragent': 'wee_slack 2.8.0'}, 50000, '', 'VIG29OKHRTIMOPA33SZ57X2ZY6SYU5H9F4RE4U4S'), {}
 258   │ Calling weechat function hook_process_hashtable with args ('url:https://api.slack.com/api/rtm.start?batch_presence_aware=1&mpim_aware=true&token=<MY_TOKEN>&nonce=
       │ EZYY', {'useragent': 'wee_slack 2.8.0'}, 50000, 'receive_httprequest_callback', 'VIG29OKHRTIMOPA33SZ57X2ZY6SYU5H9F4RE4U4S'), {}
 259   │ Calling weechat function unhook with args ('0x600003ed0370',), {}
 260   │ Calling weechat function hook_timer with args (100, 0, 0, 'handle_next', ''), {}
 261   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: RECEIVED CALLBACK with request of rtm.start id of 3e7d834dc3f31881fd0cfe986b0a84de2eff7762 and  code -2 of length 21'), {}
 262   │ Calling weechat function prnt with args ('', 'Failed connecting to slack team with token <MY-TOKEN>, will not retry after too many failed attempts. If this persists, try increasing slack_timeout. Error (code -2)
```
And after it logs this:

``` 
263   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: rtm.start failed with return_code -2. stack:\n  File "wee_slack.py", line 186, in wrapper\n    return f(*decode_from_utf8(args), **decode_from_utf8(kwargs))\n  File 
       │ "wee_slack.py", line 746, in receive_httprequest_callback\n    return_code, "".join(traceback.format_stack())\n'), {}
 264   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: RECEIVED FROM QUEUE'), {}
 265   │ Calling weechat function unhook with args ('0x600003ef31b0',), {}
 266   │ Calling weechat function hook_timer with args (10, 0, 0, 'handle_next', ''), {}
 267   │ Calling weechat function prnt with args ('0x146686e00', 'DEBUG: Max retries for Slackrequest'), {}
 268   │ Calling weechat function unhook with args ('0x600003ef1d60',), {}
 269   │ Calling weechat function hook_timer with args (100, 0, 0, 'handle_next', ''), {}
 270   │ Calling weechat function bar_item_update with args ('slack_typing_notice',), {}
```

Note that the last line (270) is being logged continuously, so far around 300 times already.

Please let me know if you need more. In the meantime I'll try to build weechat with the debug flag

**Comment by Rahmerh** [comment]
So I managed to build weechat as debug, problem is I can't install gdb due to my apple M1 max chip.. I hope these logs tell you something 😅

**Comment by Rahmerh** [comment]
So I still have no clue what happened, but I recently updated my packages (Which included python 3.10) and now it works again.

I hope it'll be fine now, I'll let you know when anything changes...

**Comment by Rahmerh** [comment]
And sidenote, the /slack register command now also works, which it didn't before..
